api/python/cellxgene_census/src/cellxgene_census/experimental/_embedding_search.py

```diff
@@ -0,0 +1,167 @@
+"""Nearest-neighbor search based on vector index of Census embeddings."""
+
+from contextlib import ExitStack
+from typing import Any, Dict, List, NamedTuple, Optional, Sequence, Tuple, cast
+
+import anndata as ad
+import numpy as np
+import numpy.typing as npt
+import pandas as pd
+import tiledb.vector_search as vs
+import tiledbsoma as soma
+
+from .._experiment import _get_experiment_name
+from .._open import DEFAULT_TILEDB_CONFIGURATION, open_soma
+from .._release_directory import CensusMirror, _get_census_mirrors
+from .._util import _uri_join
+from ._embedding import get_embedding_metadata_by_name
+
+
+class NeighborObs(NamedTuple):
+    """Results of nearest-neighbor search for Census obs embeddings."""
+
+    distances: npt.NDArray[np.float32]
+    """
+    Distances to the nearest neighbors for each query obs embedding (q by k, where q is the number
+    of query embeddings and k is the desired number of neighbors). The distance metric is
+    implementation-dependent.
+    """
+
+    neighbor_ids: npt.NDArray[np.int64]
+    """
+    obs soma_joinid's of the nearest neighbors for each query embedding (q by k).
+    """
+
+
+def find_nearest_obs(
+    embedding_name: str,
+    organism: str,
+    census_version: str,
```

Comment:
This doesn't uniquely refer to an embedding, right? It refers to the newest embedding with this name?

I think we need to be able to non-ambiguously specify an embedding here since it's really important that the embedding being queried matches what we projected our data into.